src/python/dxpy/bindings/apollo/ParseFiltersJSONbySchema.py

```diff
@@ -0,0 +1,506 @@
+            "not-between-ex",
+            "not-between-left-inc",
+            "not-between-right-inc",
+            "compare-before",
+            "compare-after",
+            "compare-within",
+        ]
+
+    def parse(self):
+        self.is_valid_json(self.schema)
+        if self.get_schema_version(self.schema) == "1.0":
+            return self.parse_v1()
+        else:
+            raise NotImplementedError
+
+    def parse_v1(self):
+        """
+        Parse input_json according to schema version 1.0, and build vizserver compound filters.
+        """
+
+        # Get the general structure of vizserver-compliant compound filter dict
+        vizserver_compound_filters = self.get_vizserver_basic_filter_structure()
+
+        # Get 'filtering_conditions' from the schema
+        all_filters = self.collect_filtering_conditions(self.schema)
+
+        # Go through the input_json (iterate through keys and values in user input JSON)
+        for filter_key, filter_values in self.input_json.items():
+            # Example: if schema is EXTRACT_ASSAY_EXPRESSION_FILTERING_CONDITIONS
+            # Then input JSON would probably contain location or annotation
+            # In this case:
+            # filter_key -> location
+            # filter_values -> list of dicts (where each dict contains "chromosome", "starting_position", "ending_position")
+            if filter_key not in all_filters:
+                self.error_handler(
+                    "No filtering condition was defined for {} in the schema.".format(
+                        filter_key
+                    )
+                )
+
+            # Get the filtering conditions for the current "key" in input_json
+            current_filters = all_filters[filter_key]
+
+            # Get the properties if any (this might be None,
+            # if so we will just execute the basic use case defined in the docstring of the class)
+            current_properties = current_filters.get("properties")
+
+            # Validate max number of allowed items
+            # if max_item_limit is defined at the top level within this key
+            # It will be later validated for each property as well
+
+            self.validate_max_item_limit(current_filters, filter_values, filter_key)
+
+            # There are several ways filtering_conditions can be defined
+            # 1. Basic use-case: no properties, just condition (see 'sample_id' in 'EXTRACT_ASSAY_EXPRESSION_FILTERING_CONDITIONS')
+            # 2. Properties defined as dict of dicts (see 'annotation' and 'expression')
+            # 3. Properties defined as list of dicts (more advanced, special use-case with complex conditional logics that needs translation)
+            # For more information see the docstring of the class
+            # The following if conditions will go through each of the aforementioned scenarios
+
+            if isinstance(current_properties, list) and isinstance(filter_values, list):
```

Comment:
I would consider breaking the `parse_v1` into smaller methods (like `parse_list_v1`, `parse_dict_v1`, etc. just for the sake of the size -to easier follow the main flow of the parsing.